src/Symfony/Component/Messenger/Exception/ChainedHandlerFailedException.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\Messenger\Envelope;
+
+class ChainedHandlerFailedException extends \RuntimeException implements ExceptionInterface
+{
+    /**
+     * @var \Throwable[]
+     */
+    private $nested;
+
+    /**
+     * @var Envelope
+     */
+    private $envelope;
+
+    public function __construct(Envelope $envelope, \Throwable ...$nested)
+    {
+        $this->envelope = $envelope;
+        $this->nested = $nested;
+        parent::__construct();
```

Comment:
Can you move this as the first call of the constructor and give it a message? If only one `$nested` then it's message, otherwise a generic message followed by the 1st `$nested` message.